internal/handlers/tigris/tigrisdb/collections.go

```diff
@@ -80,3 +91,51 @@ func (tdb *TigrisDB) CollectionExists(ctx context.Context, db, collection string
 		return false, lazyerrors.Error(err)
 	}
 }
+
+// EncodeCollName allows to have collection with / and . In the name.
+func EncodeCollName(name string) string {
+	name = strings.ReplaceAll(name, "/", "__A__")
+	return strings.ReplaceAll(name, ".", "__B__")
+}
+
+// DecodeCollName opposite of EncodeCollName.
+func DecodeCollName(name string) string {
+	name = strings.ReplaceAll(name, "__A__", "/")
+	return strings.ReplaceAll(name, "__B__", ".")
+}
+
+var schemaCache sync.Map
+
+// CacheCollectionSchema add collection schema to schema cache.
+func CacheCollectionSchema(db, collection string, schema *tjson.Schema) {
+	schemaCache.Store(db+"$$$$"+collection, schema)
+}
+
+// RefreshCollectionSchema reads and caches collection schema from the Tigris.
+func (tdb *TigrisDB) RefreshCollectionSchema(ctx context.Context, db, collection string) (*tjson.Schema, error) {
+	coll, err := tdb.Driver.UseDatabase(db).DescribeCollection(ctx, EncodeCollName(collection))
+	if err != nil {
+		return nil, err
+	}
+
+	var schema tjson.Schema
+	if err = schema.Unmarshal(coll.Schema); err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	schema.Title = DecodeCollName(collection)
+
+	schemaCache.Store(db+"$$$$"+collection, &schema)
```

Comment:
what is `$$$$`? Probably use something like `_$tigris$_`

Reply:
Just a separator. It's only in the in-memory structure. doesn't matter much what to use.

Reply:
Can be easily changed any time.